### ISSUE TITLE
Specify numpy version for dependencies

### ISSUE THREAD

**Issue by jjaraalm** [mfem/PyMFEM#60]
It looks like the binaries installed with `pip install mfem` require matching versions of numpy, however these are not specified anywhere so pip does not seem to enforce this. The result is that, in a new environment after installing mfem 4.2.0.8 and numpy 1.19.2 I get the following error:

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
RuntimeError: module compiled against API version 0xe but this version of numpy is 0xd
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-1d0a07aa8d3a> in <module>
      1 # import numpy as np
----> 2 import mfem.ser as mfem
      3 import matplotlib.pyplot as plt
      4 
      5 get_ipython().run_line_magic('matplotlib', 'widget')

~/.conda/envs/mfem/lib/python3.8/site-packages/mfem/ser.py in <module>
----> 1 from  mfem._ser.globals import *
      2 from  mfem._ser.mem_manager import *
      3 from  mfem._ser.device import *
      4 from  mfem._ser.hash import *
      5 from  mfem._ser.array import *

~/.conda/envs/mfem/lib/python3.8/site-packages/mfem/_ser/globals.py in <module>
     11 # Import the low-level C/C++ module
     12 if __package__ or "." in __name__:
---> 13     from . import _globals
     14 else:
     15     import _globals

ImportError: numpy.core.multiarray failed to import
```

For now, I'll just play around with numpy versions until it works, but it would be better if dependencies were resolved automatically.

**Comment by sshiraiwa** [comment]
Good suggestions.Binaries are built using GitHub actions. Apparently, reading a log (for example,https://github.com/mfem/PyMFEM/runs/1904511171?check_suite_focus=true), it is built either 1. 19. 5 or 1.20, depending on the version of Python. Therefore, I suppose >=1.19.5 will work. But, TBH, I am not sure why 1. 19. 2 does fail, in the sense, numpy release note does not indicate the incompatibility when 1. 19. 3-5 were released. Need to look into what is the correct way to set the version requirement. We can certainly specify the numpy version number being used to test/build.

**Comment by jjaraalm** [comment]
I'm using python 3.8 and got it work with numpy 1.20. I guess there is an API change between the 1.19 and 1.20 series.

**Comment by sshiraiwa** [comment]
Thanks. Numpy 1.20, 1.20.1 is out at the same time as I was preparing the binary release. Probably, the best is to make a new release with 1.20.1 and add the dependency information as you suggested.

**Comment by sshiraiwa** [comment]
I am closing this. Requirements.txt was added and released in PyPI as version 4.2.0.10.
This way "pip install mfem" will install numpy and scipy, if it is not installed.

Currently requirements.txt has
   numpy>=1.19.4; python_version<"3.7"
   numpy>=1.20.0; python_version>="3.7"
   scipy>=1.5.2
Note: numpy stopped supporting Python 3.6.
